debounce/simple_debouncer2_test.go

```diff
@@ -0,0 +1,105 @@
+package debounce
+
+import (
+	"sync"
+	"sync/atomic"
+	"testing"
+)
+
+// TestSimpleDebouncerRace tests SimpleDebouncer for the fact that it does not allow concurrent writing, reading.
+// So, if it`s not true `DATA RACE` will be detected.
+func TestSimpleDebouncerRace(t *testing.T) {
+	operations := 1000000
+	runs := 100
+	count := 3
+
+	d := NewSimpleDebouncer()
+	for r := 0; r < runs; r++ {
+		var wg sync.WaitGroup
+		wg.Add(count)
+
+		// The map it`s better choice for `DATA RACE` detection
+		item := make(map[int]int)
+
+		dFunc := func() {
+			for i := 0; i < operations; i++ {
+				item[1] = 1
+				tmp := item[1]
+				item[1] = tmp + 1
+			}
+		}
+
+		results := make([]bool, count)
+		for c := range results {
+			i := &results[c]
+			go func() {
+				*i = d.Debounce(dFunc)
+				wg.Done()
+			}()
+		}
+		wg.Wait()
+
+		// check results
+		if len(item) == 0 {
+			t.Fatalf("The Debounce function was not executed")
+		}
+
+		finished := 0
+		for _, done := range results {
+			if done {
+				finished++
+			}
+		}
+		if finished < 2 {
+			t.Fatalf("In one run should be finished more than 2 `Debounce` method calls, but finished %d", finished)
+		}
+	}
+}
+
+// TestDebouncerExtreme tests SimpleDebouncer in extreme conditions.
+func TestDebouncerExtreme(t *testing.T) {
```

Comment:
I think this test is not needed. It does the same as `TestSimpleDebouncerRace`

Reply:
My experiments suggest that some implementations can work normally in low-load mode, and at high load, either stuck or skips function calls.
Therefore, two different tests are needed. One for `race detect`, second for `high load` testing.